lighthouse/classes/event_properties/definitions.py

```diff
@@ -0,0 +1,213 @@
+            sample_uuids: List[str] = [  # type: ignore
+                sample[FIELD_CHERRYTRACK_LH_SAMPLE_UUID]
+                for sample in filter(
+                    lambda sample: sample[FIELD_EVENT_RUN_ID] == self.run_id_property.value,  # type: ignore
+                    filter(
+                        self.filter_pickable_samples,
+                        self.get_samples_from_source_plates(self.barcode_property.value),
+                    ),
+                )
+            ]
+            val: List[Dict[str, Any]] = list(self.get_samples_from_mongo(sample_uuids))
+            return val
+
+    def add_to_warehouse_message(self, message):
+        for sample in self.value:
+            message.add_sample_as_subject(sample)
+
+
+class UserID(EventPropertyAbstract, SimpleEventPropertyMixin):
+    def validate(self):
+        self.validate_param_not_missing(FIELD_EVENT_USER_ID)
+        self.validate_param_not_empty(FIELD_EVENT_USER_ID)
+        return self._validate
+
+    @cached_property
+    def value(self):
+        with self.retrieval_scope():
+            val = self._params.get(FIELD_EVENT_USER_ID)
+            if val is None:
+                raise RetrievalError("Unable to determine a user id")
+            return val
+
+    def add_to_warehouse_message(self, message):
+        message.set_user_id(self.value)
+
+
+class RobotSerialNumber(EventPropertyAbstract, SimpleEventPropertyMixin):
+    def validate(self):
+        self.validate_param_not_missing(FIELD_EVENT_ROBOT)
+        self.validate_param_not_empty(FIELD_EVENT_ROBOT)
+        self.validate_param_no_whitespaces(FIELD_EVENT_ROBOT)
+        return self._validate
+
+    @cached_property
+    def value(self):
+        with self.retrieval_scope():
+            return self._params.get(FIELD_EVENT_ROBOT)
+
+    def add_to_warehouse_message(self, message):
+        return None
+
+
+class RobotUUID(EventPropertyAbstract, ServiceCherrytrackMixin):
+    def __init__(self, robot_serial_number_property: RobotSerialNumber):
+        self.reset()
+        self.robot_serial_number_property = robot_serial_number_property
+
+    def validate(self):
+        return self.robot_serial_number_property.validate()
+
+    @property
+    def errors(self) -> List[str]:
+        self.validate()
+        return self._errors + self.robot_serial_number_property.errors  # type: ignore
+
+    @cached_property
+    def value(self):
+        with self.retrieval_scope():
+            val = self._get_robot_uuid()
+            if val is None:
+                raise Exception(f"Unable to determine a uuid for robot '{self.robot_serial_number_property.value}'")
+            return val
+
+    def add_to_warehouse_message(self, message):
+        message.add_subject(
+            role_type=ROLE_TYPE_ROBOT,
+            subject_type=SUBJECT_TYPE_ROBOT,
+            friendly_name=self.robot_serial_number_property.value,
+            uuid=self.value,
+        )
+
+    def _get_robot_uuid(self):
```

Comment:
The `property` decorator should be used as a getter instead of an explicit getter method.

Reply:
This method was result of a small refactor inside the class with the idea of keeping it private, not to be used as a property of the instance.